ldb/tx.go

```diff
@@ -346,3 +375,203 @@ func (db *LevelDb) FetchTxBySha(txsha *btcwire.ShaHash) ([]*btcdb.TxListReply, e
 	}
 	return replies, nil
 }
+
+// addrIndexToKey serializes the passed txAddrIndex for storage within the DB.
+// Transactions indexed by address are stored with the following format:
+//   * prefix || hash160 || blockHeight || txoffset || txlen
+// Indexes are stored purely in the key, with blank data for the actual value
+// in order to facilitate ease of iteration by their shared prefix and
+// also to allow limiting the number of returned transactions (RPC).
+// Alternatively, indexes for each address could be stored as an
+// append-only list for the stored value. However, this add unnecessary
+// overhead when storing and retrieving since the *entire* list must
+// be fetched each time.
+func addrIndexToKey(index *txAddrIndex) []byte {
+	record := make([]byte, addrIndexKeyLength, addrIndexKeyLength)
+	copy(record[:2], addrIndexKeyPrefix)
+	copy(record[2:22], index.hash160[:])
+
+	// The index itself.
+	binary.LittleEndian.PutUint64(record[22:30], uint64(index.blkHeight))
```

Comment:
I'd recommend serializing the block heights as int32.  The use of int64 for block heights elsewhere in the db is regarded as a mistake.

Should make the index slightly smaller on disk.  I don't mind changing the in-memory representation either, but then you'll probably need to make many explicit type conversions between these heights and others.